diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6ec0cb5..5aa9757 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -105,7 +105,7 @@ set(additional_libs)
 if (WIN32)
   list(APPEND ENBLEND_SOURCES win32helpers/delayHelper.h)
   list(APPEND ENFUSE_SOURCES win32helpers/delayHelper.h)
-  add_subdirectory(win32helpers)
+  # add_subdirectory(win32helpers)
   list(APPEND additional_libs getopt)
 endif()
 
diff --git a/src/enblend.cc b/src/enblend.cc
index 43ac5d4..cf9e360 100644
--- a/src/enblend.cc
+++ b/src/enblend.cc
@@ -52,8 +52,6 @@
 #include <vector>
 
 #include <getopt.h>
-extern "C" char *optarg;
-extern "C" int optind;
 
 #ifndef _MSC_VER
 #include <fenv.h>
diff --git a/src/enfuse.cc b/src/enfuse.cc
index cb7770a..653b1dc 100644
--- a/src/enfuse.cc
+++ b/src/enfuse.cc
@@ -52,8 +52,6 @@
 #include <vector>
 
 #include <getopt.h>
-extern "C" char *optarg;
-extern "C" int optind;
 
 #ifndef _MSC_VER
 #include <fenv.h>
