diff --git a/src/fixmath.h b/src/fixmath.h
index f10333f..ee29fcd 100644
--- a/src/fixmath.h
+++ b/src/fixmath.h
@@ -1791,7 +1791,11 @@ public:
         gsl_function cost = {ciecam_detail::delta_e_min_cost, &extra};
         cmsJCh opt_jch = *jch;
         double delta_e_1d = 0.0;
+#ifdef _WIN32
+        double delta_e_2d = 0.0;
+#else
         double delta_e_2d __attribute__((unused)) = 0.0;
+#endif
 
         if (EXPECT_RESULT(detail::bracket_minimum(cost, guessed_j,
                                                   0.0, std::max(static_cast<double>(Scale::MAXIMUM_LIGHTNESS), jch->J),
